src/v/cloud_storage/remote_segment.cc

```diff
@@ -1208,6 +1211,7 @@ ss::future<> hydration_loop_state::materialize() {
     }
 
     auto rs = co_await ss::when_all(fs.begin(), fs.end());
+    bool accum = true;
```

Comment:
can accum ever get false? it starts as true and it's getting only OR operations

Reply:
You're right, thanks, it should have been AND, fixed.